aws-lc-rs/src/rsa/encoding.rs

```diff
@@ -33,7 +34,126 @@ pub(in super::super) mod pkcs8 {
     }
 
     // Supports v1 and v2 encodings through a single API entry-point.
-    pub(in super::super) fn decode_der(pkcs8: &[u8]) -> Result<LcPtr<EVP_PKEY>, KeyRejected> {
-        LcPtr::try_from(pkcs8)
+    pub(in crate::rsa) fn decode_der(pkcs8: &[u8]) -> Result<LcPtr<EVP_PKEY>, KeyRejected> {
+        let mut cbs = unsafe { cbs::build_CBS(pkcs8) };
+        let key = LcPtr::new(unsafe { EVP_parse_private_key(&mut cbs) })
+            .map_err(|()| KeyRejected::unspecified())?;
```

Comment:
If we change this to returning `KeyRejected::invalid_encoding()` it fixes the Rustls integ test failure. (I was able to verify this locally.)